internal/services/costmanagement/export_resource_base.go

```diff
@@ -174,6 +189,10 @@ func (br costManagementExportBaseResource) readFunc(scopeFieldName string) sdk.R
 
 			if model := resp.Model; model != nil {
 				if props := model.Properties; props != nil {
+					if partitionData := props.PartitionData; partitionData != nil {
+						metadata.ResourceData.Set("partition_data_enabled", *partitionData)
+					}
```

Comment:
We could use `pointer.From` here to do the nil-check

```suggestion
					metadata.ResourceData.Set("partition_data_enabled", pointer.From(props.PartitionData))
```